x-pack/plugins/cases/server/client/cases/utils.ts

```diff
@@ -81,13 +81,29 @@ const getCommentContent = (comment: CommentResponse): string => {
   return '';
 };
 
-const countAlerts = (comments: CaseResponse['comments']): number =>
-  comments?.reduce<number>((total, comment) => {
-    if (comment.type === CommentType.alert || comment.type === CommentType.generatedAlert) {
-      return total + (Array.isArray(comment.alertId) ? comment.alertId.length : 1);
-    }
-    return total;
-  }, 0) ?? 0;
+interface CountAlertsInfo {
+  total: number;
+  pushed: number;
+  totalAlerts: number;
+}
+
+const countAlerts = (comments: CaseResponse['comments']): CountAlertsInfo => {
```

Comment:
I think it's a little confusing to have both `total` and `totalAlerts`. Doesn't look like we use `totalAlerts` and `pushed` other than to determine if we need to include the alerts comment. What do you think about just exposing `{ totalAlerts: number, hasUnpushedAlertComments: boolean }`?

Reply:
You are right. Is it confusing. `total` is the total comments. A comment of type alert can have more that one alert. The `totalAlerts` count that. The `pushedAlerts` is the number of comments of type alerts that have been pushed to the external service. I like the idea of `hasUnpushedAlertComments`!